### ISSUE TITLE
Error processing sketchy css background property

### ISSUE THREAD

**Issue by apexskier** [ben-eb/cssnano#129]
Given the css:

```css
.test {
  background: url('http://example.com/testing/test.png') no-repeat right #fff;
}
```

cssnano produces the error `TypeError: Cannot read property 'value' of undefined`

Depending on the build tool (gulp, postcss, cli) this is expressed in different ways, but here's an example script.

```js
var postcss = require('postcss');
var cssnano = require('cssnano');
var fs = require('fs');

postcss([cssnano()])
	.process(fs.readFileSync('example.css'), {
		from: 'example.css',
		to: 'example.min.css'
	}).then(function(result) {
		fs.writeFileSync('example.min.css', result.css);
	}).catch(function(err){
		console.warn(err);
	});
```

I realize this css isn't completely correct, but it should be dealt with, or a warning raised. Chrome actually parses it correctly, so that's why it wasn't caught before now, I think.

**Comment by TrySound** [comment]
@apexskier Can you show stack trace?

**Comment by apexskier** [comment]
```
TypeError: Cannot read property 'value' of undefined
    at /Users/cameronlittle/remitly-vagrant/html/portal/node_modules/cssnano/dist/lib/reducePositions.js:72:49
    at Array.forEach (native)
    at transform (/Users/cameronlittle/remitly-vagrant/html/portal/node_modules/cssnano/dist/lib/reducePositions.js:64:14)
    at /Users/cameronlittle/remitly-vagrant/html/portal/node_modules/postcss/lib/container.js:88:34
    at /Users/cameronlittle/remitly-vagrant/html/portal/node_modules/postcss/lib/container.js:73:26
    at Rule.each (/Users/cameronlittle/remitly-vagrant/html/portal/node_modules/postcss/lib/container.js:60:22)
    at Rule.walk (/Users/cameronlittle/remitly-vagrant/html/portal/node_modules/postcss/lib/container.js:72:21)
    at /Users/cameronlittle/remitly-vagrant/html/portal/node_modules/postcss/lib/container.js:76:32
    at Root.each (/Users/cameronlittle/remitly-vagrant/html/portal/node_modules/postcss/lib/container.js:60:22)
    at Root.walk (/Users/cameronlittle/remitly-vagrant/html/portal/node_modules/postcss/lib/container.js:72:21)
```

**Comment by apexskier** [comment]
```
TypeError: Cannot read property 'value' of undefined
    at .../node_modules/cssnano/dist/lib/reducePositions.js:72:49
    at Array.forEach (native)
    at transform (.../node_modules/cssnano/dist/lib/reducePositions.js:64:14)
    at .../node_modules/postcss/lib/container.js:88:34
    at .../node_modules/postcss/lib/container.js:73:26
    at Rule.each (.../node_modules/postcss/lib/container.js:60:22)
    at Rule.walk (.../node_modules/postcss/lib/container.js:72:21)
    at .../node_modules/postcss/lib/container.js:76:32
    at Root.each (.../node_modules/postcss/lib/container.js:60:22)
    at Root.walk (.../node_modules/postcss/lib/container.js:72:21)
```

**Comment by TrySound** [comment]
@apexskier As fast decision you can pass `reducePositions: false` option.

**Comment by apexskier** [comment]
Thanks, that solves it temporarily!

**Comment by scottzirkel** [comment]
I'm getting the same error, only I'm not sure where in my SASS the issue lies.

The `reducePositions` option fixes it. Still would be nice to know where I have an issue. Everything is rendering correctly.

**Comment by apexskier** [comment]
@scottzirkel I was able to diagnose the location by logging the declaration being processed in `reducePositions.js` to get the file, then the args to get the tokens. I searched the tokens in css output and found the problem line.

**Comment by notbrain** [comment]
Wow, just had this same issue at the same time. Just whittled it down to the first `background:url('../images/icons/check.svg') center no-repeat;` occurrence and then was like whaaaa? We have a bunch of these shortcut `background:`s all over the place. Great timing @apexskier 

@scottzirkel I'm assuming the reducePositions is trying to use shorthand for positioning where it can map things like `10px 0px 10px 0px` to `10px 0px` and the like.

**Comment by TrySound** [comment]
You can disable `reducePositions` option or use strict 3.4 version until 3.5 will be fixed.

**Comment by scottzirkel** [comment]
@apexskier Oh smart.

I got lucky and found it rather quick. I am using one `linear-gradient` background and it turns out that was it.

I have in my sass: `background: linear-gradient(to right, palette(primary, light), palette(primary), palette(primary, dark));` 

`palette(primary)` is a function that finds the primary color from the palette map and returns the value asked. In this case, light, default & dark. That is what's breaking. If I change out the palette for hex it works fine.

**Comment by mauricios** [comment]
This fixed for me using `gulp-cssnano`: 

```
var gulp = require('gulp');
var cssnano = require('gulp-cssnano');
 
gulp.task('default', function() {
    return gulp.src('./main.css')
        .pipe(cssnano({reducePositions: false}))
        .pipe(gulp.dest('./out'));
});
```

**Comment by ben-eb** [comment]
Thanks for the report, fixed in `3.5.1`. :smile:
